debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -566,6 +566,8 @@ protected void execute() {
                     // Mark the source as having completed the snapshot. This will ensure the `source` field on records
                     // are not denoted as a snapshot ...
                     source.completeSnapshot();
+                    // set the filter in the offset
+                    source.setFilterData(filters);
```

Comment:
How comes it is set here again (in addition to the task above)?

Reply:
ideally, filter data should only be set after a snapshot is taken, since, at the moment, the initial snapshot is what determines what the "real" filter info is. However, if the connector is updated to the version that contains this logic after a snapshot has already been taken, we need some method to set the filter info to what the config says it is (and hopefully it matches what the snapshot was).

So this the ideal method of setting the filter info. The previous is a fallback if the snapshot was taken without any filter data being set.

Reply:
Ok, got it.